luwak/src/main/java/uk/co/flax/luwak/presearcher/MultipassTermFilteredPresearcher.java

```diff
@@ -140,13 +142,15 @@ public Document buildQueryDocument(QueryTree querytree) {
         Document doc = new Document();
 
         for (int i = 0; i < passes; i++) {
-            Map<String, StringBuilder> fieldTerms = collectTerms(querytree);
+            Map<String, BytesRefHash> fieldTerms = collectTerms(querytree);
             debug(querytree, fieldTerms);
-            for (Map.Entry<String, StringBuilder> entry : fieldTerms.entrySet()) {
+            for (Map.Entry<String, BytesRefHash> entry : fieldTerms.entrySet()) {
                 // we add the index terms once under a suffixed field for the multipass query, and
                 // once under the plan field name for the TermsEnumTokenFilter
-                doc.add(new Field(field(entry.getKey(), i), entry.getValue().toString(), QUERYFIELDTYPE));
-                doc.add(new Field(entry.getKey(), entry.getValue().toString(), QUERYFIELDTYPE));
+                doc.add(new Field(field(entry.getKey(), i),
```

Comment:
Would this negatively affect the conventional String use case though?

I guess our issue was because we wanted a Numeric field, logically we want to send a NumericTokenStream in here, but then Luwak needs to know if a field is string or numeric....

Though if it were to know that, it would let us do NumericRangeQuery more efficiently in the presearcher?